core/src/main/scala/cats/Eval.scala

```diff
@@ -310,6 +310,30 @@ private[cats] trait EvalInstances extends EvalInstances0 {
       }
     }
 
+  implicit val catsReducibleForEval: Reducible[Eval] =
```

Comment:
I think that limiting ourselves to instances that (if they need to call `.value` at all) only call `.value` once at the end is probably a good general rule. 👍